backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -872,19 +871,8 @@ private String getRootName() {
         return root.name;
       }
 
-      @Override
-      public @Nullable String getCustomAnnotation() {
-        return null;
-      }
-
-      @Override
-      public final boolean equals(@Nullable Object other) {
-        return IBranchLayoutEntry.defaultEquals(this, other);
-      }
-
-      @Override
-      public final int hashCode() {
-        return IBranchLayoutEntry.defaultHashCode(this);
+      public BranchLayoutEntry toBranchLayoutEntry() {
+        return new BranchLayoutEntry(name, /* customAnnotation */ null, children.map(c -> c.toBranchLayoutEntry()));
```

Comment:
We have a rule:
```
* Don't write nullary lambdas in `receiver::method` notation, use explicit `() -> receiver.method()` notation instead. <br/>
  `::` notation is confusing when applied to parameterless lambdas, as it suggests a unary lambda.
```
and I know that's a one-way implication. 
But why don't we use `::` lambda version here?

Reply:
Idk, I just like it better :trollface:  (although both version are okay for unary lambdas)